### ISSUE TITLE
Component-wise scaling

### ISSUE THREAD

**Issue by caniko** [isl-org/Open3D#3936]
Currently scaling is done across all three dimensions with the same scalar for each, `mesh.scale(<scalar>)`. Can we also get support to have scale for each component, `mesh.scale((<scalar_x>, <scalar_y>, <scalar_z>)`.

**Side note (should probably have its own feature request):**
Note that this would also give us the ability to flip/mirror meshes with respect to axis of our choosing, scalar will be set to -1. Maybe add a function that wraps the scale function for flipping. Such as `mesh.mirror_x` (scalar (-1.0, 0, 0)), `mesh.mirror_y` (scalar (0, -1.0, 0)), `mesh.mirror_z` (scalar (0, 0, -1.0)).

**Comment by reyanshsolis** [comment]
I think this can already be done by using the general `transform` function, by passing it the appropriate 4x4 matrix.

**Comment by caniko** [comment]
True, however, the other functions don't support this. I was in the process of adding this feature, work got in the way. Will upload a pull request soon!
